pkg/controller/revision/revision_controller.go

```diff
@@ -281,6 +281,17 @@ func (r *ReconcileRevision) SyncReleaseManagersForRevision(revision *picchuv1alp
 				rm.Spec.Target = targetName
 				return nil
 			})
+
+			for _, rl := range rm.Status.Releases {
```

Comment:
don't we want to make sure *all* releaseManagers have expired the revision before deleting? Like increment a counter for every expired and compare counter to rm count? If equal, delete?

Reply:
Right duh, I think this is fixed.